pkg/platform/openstack/kuryr_bootstrap.go

```diff
@@ -525,6 +532,52 @@ func ensureOpenStackLbPool(client *gophercloud.ServiceClient, name, lbId string)
 	}
 }
 
+// Looks for Octavia load balancer health monitor by name and pool ID. If it does
+// not exist creates it. Will fail if multiple LB health monitors are matching all criteria.
+func ensureOpenStackLbMonitor(client *gophercloud.ServiceClient, name, poolId string) (string, error) {
+	page, err := monitors.List(client, monitors.ListOpts{
+		Name:   name,
+		PoolID: poolId,
+	}).AllPages()
+	if err != nil {
+		return "", errors.Wrap(err, "failed to get LB monitors list")
+	}
+	monitorsList, err := monitors.ExtractMonitors(page)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to extract LB monitors list")
+	}
+	if len(monitorsList) > 1 {
+		return "", errors.Errorf("found multiple LB monitors matching name %s, pool %s, cannot proceed", name, poolId)
+	} else if len(monitorsList) == 1 {
+		return monitorsList[0].ID, nil
+	} else {
+		opts := monitors.CreateOpts{
+			Name:       name,
+			PoolID:     poolId,
+			Type:       monitors.TypeTCP,
+			MaxRetries: 3,
+			Delay:      30,
+			Timeout:    10,
+
+			// FIXME(dulek): Code fragment below won't work in OpenStack Rocky as Octavia won't allow url_path on
```

Comment:
This is too bad. We configure the apiserver to report unhealthy for 30 seconds before shutting down, so that connections aren't disrupted. Definitely file a bug / issue to properly configure server health checks.

Reply:
All right, I can probably try to make this work like that at least on on Stein if I have version detection. In that case the only issue I have is access to the healthz endpoint - does API allow unauthenticated access to it?

Reply:
I believe it does - AWS does a similar thing.

Reply:
Yup, we confirmed that. :)

Reply:
I added the required version detection for Stein.